### ISSUE TITLE
Crash in AMStepScanActionController

### ISSUE THREAD

**Issue by dretrex** [acquaman/acquaman#1465]
After some discussions, it appears there is a crash bug in AMStepScanActionController.  The behaviour is typically reproduced in the generic scan tool rather than XAS, which leads @davidChevrier and myself to believe that the bug is a race condition with the file writer thread.  Since it appears less often (or never) with XAS it means that whatever time is needed for the file writer thread to do its thing is less than the time it takes to do the cleanup actions (the generic scan controller has none).

**Comment by dretrex** [comment]
@davidChevrier, I looked at this again today.  I think this has been merged in for BioXAS for the past month.  The change has either helped, or whatever else was causing their crashes seems to have abated.

To be fair, they haven't done nearly as many scans with the generic scan and have been doing XAS for a lot, but I think we can possibly make a pull request for this guy and if it rears its head again then we can open a new issue.  Thoughts?
